.github/workflows/ci.yml

```diff
@@ -0,0 +1,27 @@
+name: Tests
+
+on: [push, pull_request]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby-version:
+          - '3.1'
+          - '3.0'
+          - '2.7'
+          - '2.6'
+          - jruby-9.3
+
```

Comment:
I think, we should all Ruby versions, defined in the gemspec (`required_ruby_version`).

Also, Travis config had `ruby-head`, `jruby-head` and `rbx` with `allow_failures`.

Reply:
So GitHub actions is trickier to set up with allow failures type behavior.  It will generally show red for such failures, and as such can be confusing.  I've added a few older Ruby versions (we'll see if they're green) but considering that they're all EOLd - and Ruby 2.6 EOLs this month - I think a more appropriate move would be updating the `required_ruby_version`

Reply:
> but considering that they're all EOLd - and Ruby 2.6 EOLs this month - I think a more appropriate move would be updating the `required_ruby_version`

I definitely agree, but this process should be separate, also with new major versions releases. So… in this PR with a just new CI we should stick old practices and what gemspec guarantees.

Reply:
Well, kind of.  CI has run on this repo since October 2019.  So it's quite an assumption that those older Ruby versions run, since you're assuming that all the dependencies have maintained the same compatibility matrix.

And running against Ruby 2.0 show that assumption is not true - the current `rugged` gem doesn't build against Ruby 2.0.  Yes, it's probably possible to update the Gemfile to make it more Ruby version aware.  But spending time on that to support Ruby versions that EOLd 5+ years ago hardly seems productive.

So I'll add Ruby 2.1-2.4, and leave it there.  If the maintainers want to merge this to get CI running on the repo, they can.

Reply:
I'm almost ready to approve, but see no `-head` versions. I've checked: they supported by GH Actions.

Reply:
> So I'll add Ruby 2.1-2.4, and leave it there. If the maintainers want to merge this to get CI running on the repo, they can.

OK, in this way (I support it) this PR should be a draft, and wait for another PR with `required_ruby_version` update. But! It's not required, for sure. Just a canonical way, let's say. Your job is appreciated.

Reply:
See my comment about allow failures

Reply:
> See my comment about allow failures

OK, I got it. "can be confusing" was not a blocker for me. But, as you wish, I'm okay without head-versions, just, again, a canonical way for migration.